auditlog/audit_log_record.go

```diff
@@ -0,0 +1,100 @@
+package auditlog
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/fabric8-services/fabric8-common/convert"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// Record an audit log record to track usage of the service
+type Record struct {
+	ID          uuid.UUID   `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	CreatedAt   time.Time   `json:"created_at,omitempty"`
+	IdentityID  uuid.UUID   `sql:"type:uuid"` // TODO: or should we store the sub claim instead as a string: no need to convert it into an UUID each time
```

Comment:
I think storing the `IdentityID` is consistent with what other services do.

Reply:
@sbryzak my question in the comment is more about storing the IdentityID as a `string` rather than a `UUID`: do we really need to check if the sub claim is a `UUID` in the admin console backend service ? My point is: if we are able to verify that the token is valid and since we don't have any FK on another table in the schema, then maybe storing as a `string` is enough. WDYT?

Reply:
comment removed in 111ae7eb7d9978c56dbf97c9e2448578d23e18b6, and `IdentityID` remains an `UUID`